### DISCUSSION TITLE
Mongodb TLS connection setup

### DISCUSSION THREAD

**Discussion by b2bPrasad** [getredash/redash#6833] [Q&A] [answered]
I have a mongodb cluster with TLS enabled. I want to connect Redash to the db cluster. I setup redash using the AWS AMI option and I am trying to connect to the db server using `mongodb+srv://rs.xxxxxxxcom/<db_name>?tls=true`
I have uploaded the .crt and .pem file on to the instance and i tried the tlsCAFile and tlsCertificateKeyFile options. Note that my certificates are self-signed and my redash instance is not SSL secured
I browsed thru other related issues and I know that I am missing a few steps. I would like to know what am I missing here? And what needs to be done next.

Thank you

**Comment by justinclift** [comment]
Which version of MongoDB are you running?

Asking because the older versions of Redash won't connect to MongoDB 6.x.  We've fixed that in the latest development code, but there's not yet an official Redash release with that support.

**Reply by b2bPrasad** [reply]
I am on Mongodb 6 and Redash 10 right now. I upgraded the image by following the steps in the document.

**Reply by justinclift** [reply]
k.  That sounds like you're getting things going in the right direction.  It's still not working for you yeah?

Saying that because I was pretty sure Redash 10.1.0 wouldn't connect to MongoDB 6.  Maybe I need more coffee or something... :smile:

**Reply by b2bPrasad** [reply]
If I upgrade the database to Mongodb 7, you mean it should connect using TLS?

**Reply by justinclift** [reply]
Ahhh no.  I mean that I didn't think that Redash 10.1 would connect to MongoDB 6.  Whereas the newer development builds of Redash should connect to MongoDB 6.  At least, that's how I remember it. :smile:

**Reply by b2bPrasad** [reply]
oh okay.
If you have been able to connect at all, can you share the steps or how to progress?

**Reply by b2bPrasad** [reply]
I am using this image redash:10.0.0.b50363

**Reply by justinclift** [reply]
Sorry, I don't personally use MongoDB at all.  I was more commenting from having seen various issues filed with MongoDB 6.x, then seeing them fixed in the development code.

**Comment by b2bPrasad** [comment, marked as answer]
Alright, so I got this working by adding the certificate files in docker-compose file in the volumes tag.

**Reply by justinclift** [reply]
Good thinking. :smile:
